src/ng/directive/ngClass.js

```diff
@@ -97,15 +90,9 @@ function classDirective(name, selector) {
           oldModulo = newModulo;
         }
 
-        function ngClassOneTimeWatchAction(newClassValue) {
+        function ngClassWatchAction(newClassValue) {
           var newClassString = toClassString(newClassValue);
```

Comment:
Unfortunately we were wrong. In the case of one-time bindings the pre-interceptor value is returned up until that value passes the one-time test: https://github.com/angular/angular.js/blob/2b0c0505e222e28449829e6f4f49826b1d1ce59e/src/ng/parse.js#L1933

I've updated it to do this `toClassString` conditionally and added a comment.